### ISSUE TITLE
[Inheritance] prop options get overwritten by extended class

### ISSUE THREAD

**Issue by mohamedGamalAbuGalala** [typegoose/typegoose#292]
## Versions

![image](https://user-images.githubusercontent.com/14947957/84593609-4ca7c680-ae4d-11ea-93aa-ef273f859895.png)

## Code Example

```ts
import { prop, getModelForClass } from "@typegoose/typegoose";

class Base {
  // put default as a function if it needs to be dynamic
  @prop({ unique: true, default: () => "base" })
  UID?: string;
}

class Child extends Base {
  @prop({ unique: true, default: () => "overwritten" })
  UID?: string;
}

class GrandChild extends Child {}

const BaseModel = getModelForClass(Base);
const ChildModel = getModelForClass(Child);
const GrandChildModel = getModelForClass(GrandChild);

// # Working

const b = new BaseModel();
const c = new ChildModel();
const gc = new GrandChildModel();

console.log(b.UID);          // base

console.log(c.UID);         // overwritten

console.log(gc.UID);       // base ??  Should it print "overwritten" or I miss something?
```

## What should happen?

The GrandChildModel `UID` default value should be from  `Child`, not from the `base`.

## Do you know *why* it happens?

*no*

**Comment by hasezoey** [comment]
Confirmed, happening even with an prop

```ts
// NodeJS: 14.4.0
// MongoDB: 4.2-bionic (Docker)
import { getModelForClass, prop } from "@typegoose/typegoose"; // @typegoose/typegoose@7.2.0
import * as mongoose from "mongoose"; // mongoose@5.9.18 @types/mongoose@5.7.21

class Base {
  // put default as a function if it needs to be dynamic
  @prop({ unique: true, default: () => "base" })
  public UID?: string;
}

class Child extends Base {
  @prop({ unique: true, default: () => "overwritten" })
  public UID?: string;
}

class GrandChild extends Child {
  @prop()
  public something?: string;
}

const BaseModel = getModelForClass(Base);
const ChildModel = getModelForClass(Child);
const GrandChildModel = getModelForClass(GrandChild);

(async () => {
  await mongoose.connect(`mongodb://localhost:27017/`, { useNewUrlParser: true, dbName: "verifyMASTER", useCreateIndex: true, useUnifiedTopology: true });

  const b = new BaseModel();
  const c = new ChildModel();
  const gc = new GrandChildModel();

  console.log(b.UID);

  console.log(c.UID);

  console.log(gc.UID);

  await mongoose.disconnect();
})();
```

Note: at first i thought it might be a problem related that `GrandChild` is empty, which causes some other issues

**Comment by hasezoey** [comment]
This problem will be fixed by https://github.com/typegoose/typegoose/pull/243, the pr cannot be merged until https://github.com/Automattic/mongoose/pull/8897 is released in mongoose, because of https://github.com/Automattic/mongoose/issues/8892

TL;DR: #243 would fix it, but would break virtuals

**Comment by hasezoey** [comment]
the current status is that this issue cant be fixed without majorly breaking(/changing) other things before mongoose releases 6.0

**Comment by hasezoey** [comment]
closing because fixed in beta & master (9.0.0)
